Yoast/wordpress-seo
# Add postSettingsModal actions

## Context


* We want the metabox modal to have some sweet content :dancers: 

## Summary


This PR can be summarized in the following changelog entry:

* Adds content sections to the modal.
* Adds styling to the modal.
* Refactors two usages of `withSelect/withDispatch` alongside components into `container` and `component` structure.
* Groups and moves all CSS for our Modal.
* Removes the `Collapsibles` from the `AdvancedTab` and the `SchemaTab` and adds them to where they are rendered.

## Relevant technical choices:

* The `getPostTypeName` function only only returns `post` or `page` per https://yoast.slack.com/archives/CSYSU835E/p1596703391029800?thread_ts=1596610996.006100&cid=CSYSU835E :heavy_check_mark: 

## Test instructions

This PR can be tested by following these steps:
1. Check that the `Modal` looks like in the spec. There are two differences:
	1. The `Collapsible` headings are grey instead of purple. This is per https://yoast.slack.com/archives/CSYSU835E/p1597394920006400?thread_ts=1597394161.006300&cid=CSYSU835E.
	2. Only the mobile variant has been implemented. This scales to a larger "mobile" modal on desktop displays. If you still see collapsibles in desktop view: that is the desktop "mobile" modal in action.
2. Check that functionality inside the modal works as expected, this means:
	1. Change settings
	2. Close the modal and save the post
	3. Reload the page
	4. Open the modal again
	5. Are the settings persisted?

	There are two known errors:
	1. [The modal closes after the media library modal closes](https://yoast.atlassian.net/browse/P1-134)
	2. [Meta robots advanced does not display correctly](https://yoast.atlassian.net/browse/P1-135)
3. Check that the `AdvancedTab` and the `SchemaTab` still display as expected in the metabox / sidebar (only the `SchemaTab` is rendered in the sidebar).
4. Make sure that the `PrimaryTaxonomyFilter` still works.
5. Make sure that the Content Analysis still works.

## UI changes

## Documentation

## Quality assurance


Fixes P1-106 and P1-107
